pkgs/build-support/testers/default.nix

```diff
@@ -15,4 +15,33 @@
         echo "$output" >&2 && exit 1
       fi
     '';
+
+  testGraphical =
+    { package
+    , command ? "${package.meta.mainProgram or package.pname or package.name}"
+    , expectedText ? "File" # Very very common in menu bars.
+    }: pkgs.nixosTest ({ pkgs, ... }: {
+      name = "${package.name}-test-graphical";
+
+      nodes.machine = { ... }: {
+        imports = [ ../../../nixos/tests/common/wayland-cage.nix ];
+        environment.systemPackages = [ package ];
+        services.cage.program = "${package}/bin/${command}";
+      };
+
+      enableOCR = expectedText != null;
+
+      testScript = ''
+        @polling_condition
+        def cage_running():
+          machine.require_unit_state("cage-tty1.service", "active")
+
+
+        machine.wait_for_unit("cage-tty1.service")
+        with cage_running:
+            machine.wait_until_succeeds("pgrep -f ${command}")
+            machine.wait_for_text("${expectedText}", verbose=True)
```

Comment:
This needs escaping. We should to solve this "once and for all" with a new `variables` test option, that takes any json object and turns it into variables that are brought into the script's scope.